modin/config/envvars.py

```diff
@@ -0,0 +1,224 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import os
+from textwrap import dedent
+import warnings
+from packaging import version
+
+from .pubsub import Parameter, _TYPE_PARAMS
+
+
+class EnvironmentVariable(Parameter, type=str, abstract=True):
+    """
+    Base class for environment variables-based configuration
+    """
+
+    varname: str = None
+
+    @classmethod
+    def _get_raw_from_config(cls) -> str:
+        return os.environ[cls.varname]
+
+    @classmethod
+    def get_help(cls) -> str:
```

Comment:
Maybe add some short example of help string?

Reply:
I don't see a need for it. The output is provided in PR description if you expand the spoiler, no need IMHO to insert it into the source.